airbyte-integrations/connectors/source-greenhouse/source_greenhouse/greenhouse.yaml

```diff
@@ -35,6 +35,14 @@ definitions:
       page_token_option:
         inject_into: "path"
       url_base: "*ref(definitions.requester.url_base)"
+  no-pagination-retriever:
+    type: SimpleRetriever
```

Comment:
nit: the requester could be defined here so only the path needs to be defined in the stream definitions

Reply:
Do you suggest I do something like this
```
 no-pagination-retriever:
    type: SimpleRetriever
    name: "{{ options['name'] }}"
    primary_key: "{{ options['primary_key'] }}"
    record_selector:
      $ref: "*ref(definitions.selector)"
    requester:
      $ref: "*ref(definitions.requester)"
    paginator:
      type: NoPagination

 tags_stream:
    $ref: "*ref(definitions.base_stream)"
    $options:
      name: "tags"
      path: "tags/candidate"
    retriever:
      $ref: "*ref(definitions.no-pagination-retriever)"

 user_permissions_stream:
    $options:
      name: "user_permissions"
    primary_key: "id"
    schema_loader:
      $ref: "*ref(definitions.schema_loader)"
    retriever:
      $ref: "*ref(definitions.no-pagination-retriever)"
      requester:
        path: "users/{{ stream_slice.parent_id }}/permissions/jobs"
      stream_slicer:
        type: SubstreamSlicer
        parent_stream_configs:
          - stream: "*ref(definitions.users_stream)"
            parent_key: "id"
            stream_slice_field: "parent_id"

```

Is this a valid configuration? can I override the requester in the user_permissions_stream without specifying a reference to the requester being overridden with  $ref: "*ref(definitions.requester)"? Is it automatically picked up from the retriever?

Reply:
You still need to reference it if you overload it. However, you can define the `path` in a top-level `$options` block and it'll get propagated 


```
 no-pagination-retriever:
    type: SimpleRetriever
    name: "{{ options['name'] }}"
    primary_key: "{{ options['primary_key'] }}"
    record_selector:
      $ref: "*ref(definitions.selector)"
    requester:
      $ref: "*ref(definitions.requester)"
    paginator:
      type: NoPagination

 tags_stream:
    $ref: "*ref(definitions.base_stream)"
    $options:
      name: "tags"
      path: "tags/candidate"
    retriever:
      $ref: "*ref(definitions.no-pagination-retriever)"

 user_permissions_stream:
    $options:
      name: "user_permissions"
      path: "users/{{ stream_slice.parent_id }}/permissions/jobs"
    primary_key: "id"
    schema_loader:
      $ref: "*ref(definitions.schema_loader)"
    retriever:
      $ref: "*ref(definitions.no-pagination-retriever)"
      stream_slicer:
        type: SubstreamSlicer
        parent_stream_configs:
          - stream: "*ref(definitions.users_stream)"
            parent_key: "id"
            stream_slice_field: "parent_id"
```